plugins/woocommerce/includes/abstracts/abstract-wc-order.php

```diff
@@ -638,7 +638,7 @@ public function set_status( $new_status ) {
 			}
 
 			// If the old status is set but unknown (e.g. draft) assume its pending for action usage.
-			if ( $old_status && ! in_array( 'wc-' . $old_status, $this->get_valid_statuses(), true ) && ! in_array( $old_status, $status_exceptions, true ) ) {
+			if ( $old_status && ( ( ! in_array( 'wc-' . $old_status, $this->get_valid_statuses(), true ) && ! in_array( $old_status, $status_exceptions, true ) ) || 'auto-draft' === $old_status ) ) {
```

Comment:
Adding the new condition first will allow skipping the `in_array` scans when it's an `auto-draft`

```suggestion
			if ( $old_status && ( 'auto-draft' === $old_status || ( ! in_array( 'wc-' . $old_status, $this->get_valid_statuses(), true ) && ! in_array( $old_status, $status_exceptions, true ) ) ) {
```

Reply:
Makes sense. Implemented in 3c27123aa2cf83c8d579846e5ebd3bb733c368aa.